### ISSUE TITLE
TS4094 error occurred, even if the parent has a protected member (`declaration: true`)

### ISSUE THREAD

**Issue by karikera** [microsoft/TypeScript#47186]
# Bug Report

* Similar issue: #17293
### 🔎 Search Terms

ts4094 parent
ts4094 extends

### 🕗 Version & Regression Information
4.6.0-dev.20211218
- This is the behavior in every version I tried, and I reviewed the FAQ for entries about _________

### ⏯ Playground Link
[[Playground link with relevant code](https://www.typescriptlang.org/dev/bug-workbench/?ts=4.6.0-dev.20211216#code/LAKAxgNghgzjAEAxA9s+BvU9vwA4CdkAXAUzFIBN4p4BeeABgG5QBfUMZAOxiOrvhcSAd3iRYCEgA9SXCghRp0rJkA)]
* The playground doesn't show this error. it needs `"declaration": true`

### 💻 Code

```ts

class Foo {
    protected a = 0;
}
const a = new class extends Foo {};

```

### 🙁 Actual behavior

ts4094, Property 'a' of exported class expression may not be private or protected.ts(4094)

### 🙂 Expected behavior

no error.
'a' can be accessed by the other child classes.

**Comment by MartinJohns** [comment]
> The playground doesn't show this error. it needs `"declaration": true`

It does, when you add `// @declaration: true`.

**Comment by MartinJohns** [comment]
I don't see any difference to #17293.

**Comment by fatcerberus** [comment]
I believe this could be fixed if #41587 were implemented.  Right now it looks like a design limitation, as explained in #17293.

**Comment by karikera** [comment]
> I don't see any difference to #17293.

I felt it little different but it seems almost same.
